File: tests/csharp/Reference.Tests.CSharp.MSTest/NLogUnitTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using NLog;
using NLog.Config;

namespace Reference.Tests.CSharp.MSTest
{
    /// <summary>
    /// NLog UnitTest
    /// </summary>
    [TestClass]
    public class NLogUnitTest
    {
        private Logger _logger;

        /// <summary>
        /// Logger configuration
        /// </summary>
        [TestInitialize]
        public void StartUp()
        {
            LogManager.Configuration = new LoggingConfiguration();
            _logger = LogManager.GetCurrentClassLogger();
        }

        /// <summary>
        /// NLog test
        /// </summary>
        [TestMethod]
        public void NLogExample()
        {
            _logger.Trace("Trace in NLog");
            _logger.Debug("Debug in NLog");
            _logger.Info("Info in NLog");
            _logger.Warn("Warn in NLog");
            _logger.Error("Error in NLog");
            _logger.Fatal("Fatal in NLog");

            try
            {
                throw new Exception("Sample exception");
            }
            catch (Exception ex)
            {
                _logger.Error(ex, "Exception error.");
            }

            _logger.Info("Structured log: {logStruct}", new { Id = 6, Name = "Jacket", Color = "Orange" });
        }
    }
}

File: tests/vb/Reference.Tests.VB.MSTest/NLogUnitTest.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports NLog
Imports NLog.Config

''' <summary>
''' NLog UnitTest
''' </summary>
<TestClass>
Public Class NLogUnitTest
    Private _logger As Logger

    ''' <summary>
    ''' Logger configuration
    ''' </summary>
    <TestInitialize>
    Public Sub StartUp()
        LogManager.Configuration = New LoggingConfiguration()
        _logger = LogManager.GetCurrentClassLogger()
    End Sub

    ''' <summary>
    ''' NLog test
    ''' </summary>
    <TestMethod>
    Public Sub NLogExample()
        _logger.Trace("Trace in NLog")
        _logger.Debug("Debug in NLog")
        _logger.Info("Info in NLog")
        _logger.Warn("Warn in NLog")
        _logger.Error("Error in NLog")
        _logger.Fatal("Fatal in NLog")

        Try
            Throw New Exception("Sample exception")
        Catch ex As Exception
            _logger.Error(ex, "Exception error.")
        End Try

        _logger.Info("Structured log: {logStruct}", New With {.Id = 6, .Name = "Jacket", .Color = "Orange"})
    End Sub
End Class
